components/script/dom/webglrenderingcontext.rs

```diff
@@ -2285,6 +2290,33 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
         }
     }
 
+    #[allow(unsafe_code)]
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.6
+    unsafe fn GetFramebufferAttachmentParameter(&self, _cx: *mut JSContext,
+                                         target: u32, attachment: u32, pname: u32) -> JSVal
+    {
+        // Check if currently bound framebuffer is non-zero as per spec.
+        if let None = self.bound_framebuffer.get() {
+            self.webgl_error(InvalidOperation);
+            return NullValue();
+        }
```

Comment:
It's probably easier and faster if we do the validation here, to keep it all in the same place.

The WebGL thread should assert that glGetError returns GL_NO_ERROR in debug builds, in case we've messed up.

Does that sound fine?

Reply:
Though now I see it we're very inconsistent about it, I think it's nicer. wdyt? (also, @jdm wdyt?)

Reply:
Yes, it's a bit inconsistent in the code base. I noticed that usually on the receiver channel side we validate input that we pass to GLES and on the sender side (here) we check what GLES returned. If it returns something we don't expect, we panic.

Validating everything here wouldn't be any harder, we could do that if that's faster.

Reply:
Looking at it again today, I think it will look nicer, I agree. If that's faster too, then it's a win-win. We could move all validation for other APIs here too as a follow-up.

@jdm What do you think? Sounds good?